packages/neuron-wallet/src/services/sync-progress.ts

```diff
@@ -0,0 +1,81 @@
+import { getConnection, In, Not } from 'typeorm'
+import SyncProgress from 'database/chain/entities/sync-progress'
+import { scriptToHash } from '@nervosnetwork/ckb-sdk-utils'
+import { HexString } from '@ckb-lumos/base'
+
+export default class SyncProgressService {
+  static async resetSyncProgress(
+    params: { script: CKBComponents.Script; scriptType: CKBRPC.ScriptType; walletId: string }[]
+  ) {
+    await getConnection()
+      .getRepository(SyncProgress)
+      .createQueryBuilder()
+      .insert()
+      .orIgnore()
+      .values(params.map(v => SyncProgress.fromObject(v)))
+      .execute()
+    const walletIds = [...new Set(params.map(v => v.walletId))]
+    await getConnection()
+      .createQueryBuilder()
+      .update(SyncProgress)
+      .set({ delete: true })
+      .where({ walletId: Not(In(walletIds)) })
```

Comment:
I didn't get this one, the method looks to reset progress of scripts belonging to specific wallets, but the wallets other than the specified ones are deleted?

Reply:
Yeah, the wallets that are not in sync will be set to delete true. But it may be better to split these into two functions.
And the two functions will be called when init sync.